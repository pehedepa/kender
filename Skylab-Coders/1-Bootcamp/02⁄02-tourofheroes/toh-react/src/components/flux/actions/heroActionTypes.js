const heroActionTypes = {
  ADD_HERO: 'ADD_HERO',
  DELETE_HERO: 'DELETE_HERO',
  UPDATE_HERO: 'UPDATE HERO',
  LOAD_HEROES: 'LOAD HEROES',
};

export default heroActionTypes;
