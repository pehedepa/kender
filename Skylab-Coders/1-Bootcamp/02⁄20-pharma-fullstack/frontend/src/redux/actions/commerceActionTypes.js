export default {
  LOAD_ITEMS: 'LOAD_ITEMS',
  ADD_ITEM_TO_CART: 'ADD_ITEM_TO_CART'
};
