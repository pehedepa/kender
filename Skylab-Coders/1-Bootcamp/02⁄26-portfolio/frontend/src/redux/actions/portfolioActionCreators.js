import axios from 'axios';
import portfolioActionTypes from './portfolioActionTypes';

const api = 'http://localhost:5000/api/portfolio';

export default function getGenericData() {
  const generic = '/generic';
  return async (dispatch) => {
    const { data } = await axios(api + generic);
    dispatch({
      type: portfolioActionTypes.GET_GENERIC,
      data
    });
  };
}

export function getTradeSkillsData() {
  const tradeSkills = '/tradeskills';
  return async (dispatch) => {
    const { data } = await axios(api + tradeSkills);
    dispatch({
      type: portfolioActionTypes.GET_TRADESKILLS,
      data
    });
  };
}
