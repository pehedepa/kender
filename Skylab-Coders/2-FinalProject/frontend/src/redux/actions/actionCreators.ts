import axios from 'axios'
import actionTypes from './actionTypes'
import { loginURL, registerURL, productsURL, usersURL, logoutURL } from '../../constants'
import { Dispatch } from 'react'

export function login (email: string, password: string): Promise<any> {
  const data = { email, password }
  return async function dispatchInfo (dispatch) {
    try {
      const response = await axios.post(loginURL, data)
      dispatch({
        type: actionTypes.LOGIN_USER,
        payload: response
      })
    } catch (error) {
      dispatch({
        type: actionTypes.LOGIN_ERROR,
        payload: 401
      })
    }
  }
}

export function createAcc (email: string, password: string, location: string, name: string): Promise<any> {
  const data = { email, password, location, name }
  return async function dispatchInfo (dispatch) {
    try {
      const response = await axios.post(registerURL, data)
      dispatch({
        type: actionTypes.ADD_USER,
        payload: response.status
      })
    } catch (error) {
      dispatch({
        type: actionTypes.ADD_USER_ERROR,
        payload: 401
      })
    }
  }
}

export function uploadProduct (name: string, price: number, description: string, image: string, user: string, category: string, ageRange: string, keywords: any[]): Promise<any> {
  const data = { name, price, description, image, user, category, ageRange, keywords }
  return async function dispatchInfo (dispatch) {
    try {
      const response = await axios.post(productsURL, data)
      dispatch({
        type: actionTypes.UPLOAD_PRODUCT,
        payload: response.status
      })
    } catch (error) {
      dispatch({
        type: actionTypes.UPLOAD_PRODUCT,
        payload: 401
      })
    }
  }
}

export function getAllProducts (): Promise<any> {
  return async function dispatchInfo (dispatch) {
    try {
      const response = await axios.get(productsURL)
      dispatch({
        type: actionTypes.GET_ALL_PRODUCTS,
        payload: response.data
      })
    } catch (error) {
      dispatch({
        type: actionTypes.GET_ALL_PRODUCTS_ERROR,
        payload: 401
      })
    }
  }
}

export function uploadProfile (_id, photo, name, address, cc, location): Promise<any> {
  const data = { _id, photo, name, address, cc, location }
  return async function dispatchInfo (dispatch) {
    try {
      const response = await axios.put(usersURL, data)
      dispatch({
        type: actionTypes.UPDATE_PROFILE,
        payload: response
      })
    } catch (error) {
      dispatch({
        type: actionTypes.UPDATE_PROFILE,
        payload: 401
      })
    }
  }
}

export function logout (email, password): Promise<any> {
  const data = { email, password }
  return async function dispatchInfo (dispatch) {
    try {
      await axios.post(logoutURL, data)
      dispatch({
        type: actionTypes.LOG_OUT,
        payload: null
      })
    } catch (error) {
      dispatch({
        type: actionTypes.LOG_OUT,
        payload: null
      })
    }
  }
}

export function deleteProduct (_id) {
  return async function dispatchInfo (dispatch) {
    try {
      await axios.delete(`${productsURL}/${_id}`)
      dispatch({
        type: actionTypes.DELETE_PRODUCT
      })
    } catch (error) {
      dispatch({
        type: actionTypes.DELETE_PRODUCT,
        payload: null
      })
    }
  }
}

export function applyFilter (data: string) : { type: string, payload: string } {
  return {
    type: actionTypes.FILTER_PRODUCTS,
    payload: data
  }
}

export function cleanFilters (): { type: string, payload: null } {
  return {
    type: actionTypes.CLEAN_FILTERS,
    payload: null
  }
}

export function cleanRegister (): { type: string } {
  return {
    type: actionTypes.CLEAN_REGISTER_STATUS
  }
}

export function cleanLogin (): { type: string } {
  return {
    type: actionTypes.CLEAN_LOGIN_STATUS
  }
}

export function cleanProductUploaded (): { type: string } {
  return {
    type: actionTypes.CLEAN_UPLOAD_PRODUCT
  }
}
