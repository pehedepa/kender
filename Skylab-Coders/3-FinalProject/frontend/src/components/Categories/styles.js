import { StyleSheet } from 'react-native'
import { ThemeVars } from '../../styles/theme.styles'

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    width: '100%',
    backgroundColor: '#ffffff',
    borderBottomColor: ThemeVars.BORDER_COLOR,
    borderBottomWidth: 1,
    paddingHorizontal: ThemeVars.CONTAINER_PADDING,
    paddingVertical: 15
  },
  icon: {
    width: 45,
    height: 45,
    resizeMode: 'contain'
  },
  text: {
    textAlign: 'center',
    paddingTop: 5,
    fontFamily: ThemeVars.FONT_FAMILY_LIGHT
  }
})

export default styles
