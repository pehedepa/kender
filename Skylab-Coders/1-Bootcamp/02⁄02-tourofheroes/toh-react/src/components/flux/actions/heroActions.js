import dispatcher from '../dispatcher/dispatcher';
import heroActionTypes from './heroActionTypes';

export function saveHero(hero) {
  debugger;
  const action = {
    type: hero.id ? heroActionTypes.UPDATE_HERO : heroActionTypes.ADD_HERO,
    hero,
  };

  dispatcher.dispatch(action);
}

export function deleteHero({ id }) {
  debugger;
  const action = {
    type: heroActionTypes.DELETE_HERO,
    heroId: id,
  };

  dispatcher.dispatch(action);
}

export function loadHero(heroId) {
  const action = {
    type: heroActionTypes.LOAD_HERO,
    heroId,
  };

  dispatcher.dispatch(action);
}
