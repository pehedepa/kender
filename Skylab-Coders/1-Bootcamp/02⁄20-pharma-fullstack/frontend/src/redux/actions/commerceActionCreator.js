import axios from 'axios';
import commerceActionTypes from './commerceActionTypes';

const api = 'http://localhost:5000/api/commerce';

export default function getItems() {
  return async (dispatch) => {
    const { data } = await axios(api);
    dispatch({
      type: commerceActionTypes.LOAD_ITEMS,
      data
    });
  };
}

export function pushItemToCart(name, _id, url, stock, price) {
  const payload = {
    name,
    _id,
    url,
    stock,
    price
  };

  return (dispatch) => {
    dispatch({
      type: commerceActionTypes.ADD_ITEM_TO_CART,
      data: payload
    });
  };
}
