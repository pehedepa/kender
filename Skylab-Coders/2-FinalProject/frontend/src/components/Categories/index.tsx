import React from 'react'
import { View, Image, Text } from 'react-native'
import { TouchableOpacity } from 'react-native-gesture-handler'
import { connect } from 'react-redux'
import { toys, clothes, other, books } from '../../constants'
import { bindActionCreators } from 'redux'
import { applyFilter, getAllProducts, cleanFilters } from '../../redux/actions/actionCreators'
import styles from './styles'

function Categories ({ productList, actions }: any): any {
  function filterByCategory (category: string | null) {
    const resultArray: any[] = productList.filter(item => item.category === category)
    if (category === null) {
      actions.cleanFilters()
    } else {
      actions.applyFilter(resultArray)
    }
  }

  return (
    <View style={styles.container}>
      <TouchableOpacity
        testID='Toys'
        onPress={() => filterByCategory(toys)}
      >
        <Image
          style={styles.icon}
          source={require('../../assets/icons/cat_toys.png')}
        />
        <Text style={styles.text}>Toys</Text>
      </TouchableOpacity>

      <TouchableOpacity
        testID='Clothes'
        onPress={() => filterByCategory(clothes)}
      >
        <Image
          style={styles.icon}
          source={require('../../assets/icons/cat_clothes.png')}
        />
        <Text style={styles.text}>Clothes</Text>
      </TouchableOpacity>

      <TouchableOpacity
        testID='Books'
        onPress={() => filterByCategory(books)}
      >
        <Image
          style={styles.icon}
          source={require('../../assets/icons/cat_books.png')}
        />
        <Text style={styles.text}>Books</Text>
      </TouchableOpacity>

      <TouchableOpacity
        testID='Other'
        onPress={() => filterByCategory(other)}
      >
        <Image
          style={styles.icon}
          source={require('../../assets/icons/cat_accessories.png')}
        />
        <Text style={styles.text}>Other</Text>
      </TouchableOpacity>

      <TouchableOpacity
        testID='All'
        onPress={() => filterByCategory(null)}
      >
        <Image
          style={styles.icon}
          source={require('../../assets/icons/filter.png')}
        />
        <Text style={styles.text}>See All</Text>
      </TouchableOpacity>
    </View>
  )
}

function mapStateToProps (state) {
  return { productList: state.productList }
}

function mapDispatchToProps (dispatch) {
  return {
    actions: bindActionCreators({ applyFilter, cleanFilters, getAllProducts }, dispatch)
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Categories)
