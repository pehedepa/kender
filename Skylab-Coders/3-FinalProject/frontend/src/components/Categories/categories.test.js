import React from 'react'
import Categories from './index'
import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import * as actions from '../../redux/actions/actionCreators'
import { render, fireEvent } from '@testing-library/react-native'
import thunk from 'redux-thunk'

describe('Given a Categories function', () => {
  const mockStore = configureStore([thunk])
  let store
  beforeEach(() => {
    jest.spyOn(actions, 'applyFilter').mockReturnValueOnce({ type: '' })
    jest.spyOn(actions, 'cleanFilters').mockReturnValueOnce({ type: '' })
    jest.spyOn(actions, 'getAllProducts').mockReturnValueOnce({ type: '' })

    store = mockStore({
      productList: [
        { category: 'Toys' },
        { category: 'Books' },
        { category: 'Other' }
      ]
    })
  })

  describe('When we invoke Categories function', () => {
    test('Then the render should match its snaptshot', () => {
      const tree = render(<Provider store={store}><Categories></Categories></Provider>)

      expect(tree).toMatchSnapshot()
    })
  })

  describe('When we press Toys category button and filterByCategory gets called', () => {
    test('Then TOYS argument should invoke actions.applyFilter', () => {
      const { getByTestId } = render(<Provider store={store}><Categories></Categories></Provider>)

      fireEvent.press(getByTestId('Toys'))

      expect(actions.applyFilter).toHaveBeenCalled()
    })
  })

  describe('When we press Clothes category button and filterByCategory gets called', () => {
    test('Then CLOTHES argument should invoke actions.applyFilter', () => {
      const { getByTestId } = render(<Provider store={store}><Categories></Categories></Provider>)

      fireEvent.press(getByTestId('Clothes'))

      expect(actions.applyFilter).toHaveBeenCalled()
    })
  })

  describe('When we press Books category button and filterByCategory gets called', () => {
    test('Then BOOKS argument should invoke actions.applyFilter', () => {
      const { getByTestId } = render(<Provider store={store}><Categories></Categories></Provider>)

      fireEvent.press(getByTestId('Books'))

      expect(actions.applyFilter).toHaveBeenCalledWith()
    })
  })

  describe('When we press Other category button and filterByCategory gets called', () => {
    test('Then OTHER argument should invoke actions.applyFilter', () => {
      const { getByTestId } = render(<Provider store={store}><Categories></Categories></Provider>)

      fireEvent.press(getByTestId('Other'))

      expect(actions.applyFilter).toHaveBeenCalled()
    })
  })

  describe('When we press All category button and filterByCategory gets called', () => {
    test('Then NULL argument should invoke actions.cleanFilters', () => {
      const { getByTestId } = render(<Provider store={store}><Categories></Categories></Provider>)

      fireEvent.press(getByTestId('All'))

      expect(actions.cleanFilters).toHaveBeenCalled()
    })
  })
})
