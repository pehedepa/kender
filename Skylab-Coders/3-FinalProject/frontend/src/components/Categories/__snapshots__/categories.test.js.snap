// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Categories function When we invoke Categories function Then the render should match its snaptshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderBottomColor": "#e0e0e0",
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingHorizontal": 25,
      "paddingVertical": 15,
      "width": "100%",
    }
  }
>
  <RNGestureHandlerButton
    collapsable={false}
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    rippleColor={0}
    testID="Toys"
  >
    <View
      accessible={true}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 45,
            "resizeMode": "contain",
            "width": 45,
          }
        }
      />
      <Text
        style={
          Object {
            "fontFamily": "MontserratLight",
            "paddingTop": 5,
            "textAlign": "center",
          }
        }
      >
        Toys
      </Text>
    </View>
  </RNGestureHandlerButton>
  <RNGestureHandlerButton
    collapsable={false}
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    rippleColor={0}
    testID="Clothes"
  >
    <View
      accessible={true}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 45,
            "resizeMode": "contain",
            "width": 45,
          }
        }
      />
      <Text
        style={
          Object {
            "fontFamily": "MontserratLight",
            "paddingTop": 5,
            "textAlign": "center",
          }
        }
      >
        Clothes
      </Text>
    </View>
  </RNGestureHandlerButton>
  <RNGestureHandlerButton
    collapsable={false}
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    rippleColor={0}
    testID="Books"
  >
    <View
      accessible={true}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 45,
            "resizeMode": "contain",
            "width": 45,
          }
        }
      />
      <Text
        style={
          Object {
            "fontFamily": "MontserratLight",
            "paddingTop": 5,
            "textAlign": "center",
          }
        }
      >
        Books
      </Text>
    </View>
  </RNGestureHandlerButton>
  <RNGestureHandlerButton
    collapsable={false}
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    rippleColor={0}
    testID="Other"
  >
    <View
      accessible={true}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 45,
            "resizeMode": "contain",
            "width": 45,
          }
        }
      />
      <Text
        style={
          Object {
            "fontFamily": "MontserratLight",
            "paddingTop": 5,
            "textAlign": "center",
          }
        }
      >
        Other
      </Text>
    </View>
  </RNGestureHandlerButton>
  <RNGestureHandlerButton
    collapsable={false}
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    rippleColor={0}
    testID="All"
  >
    <View
      accessible={true}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 45,
            "resizeMode": "contain",
            "width": 45,
          }
        }
      />
      <Text
        style={
          Object {
            "fontFamily": "MontserratLight",
            "paddingTop": 5,
            "textAlign": "center",
          }
        }
      >
        See All
      </Text>
    </View>
  </RNGestureHandlerButton>
</View>
`;
